kairon/shared/data/processor.py

```
@@ -254,6 +254,211 @@
+
+        llm_prompts = config.get("llm_prompts", [])
+        for prompt in llm_prompts:
+            if prompt.get('source') == "slot" and prompt.get('data') == slot:
+                return True
+
+        set_slots = config.get("set_slots", [])
+        for set_slot in set_slots:
+            if set_slot.get('name') == slot:
+                return True
+
+        return False
```

Ruff

Use `return any(set_slot.get("name") == slot for set_slot in set_slots)` instead of `for` loop

Replace with `return any(set_slot.get("name") == slot for set_slot in set_slots)`

(SIM110)